pkg/solace/resource/destination.go

```diff
@@ -171,3 +171,134 @@ func QueueNonDurableExclusiveAnonymous() *Queue {
+	// messages are received, passing those instead and passing no more cached messages.
+	LiveCancelsCached
+
+	// CachedFirst provides a configuration for passing only cached messages to the application, before passing the received live messages.
+	// The live messages passed to the application thereof this configuration can be received as early as when the cache request is sent
+	// by the API, and are enqueued until the cache response is received and its associated cached messages, if available, are passed to
+	// the application.
+	CachedFirst
+
+	// CachedOnly provides a configuration for passing only cached messages and no live messages to the application.
+	CachedOnly
+)
+
+// CachedMessageSubscriptionRequest provides an interface through which cache request configurations can be constructed. These
+// configurations can then be passed to a call to a [RequestCached] interface method to request cached data. Refer to each of the below
+// factory methods for details on what configuration they provide.
+type CachedMessageSubscriptionRequest interface {
+	// GetCacheName retrieves the name of the cache.
+	GetCacheName() string
+
+	// GetName retrieves the name of the topic subscription.
+	GetName() string
+
+	GetSubscription() *TopicSubscription
+
+	GetCacheAccessTimeout() int32
+
+	GetMaxCachedMessages() int32
+
+	GetCachedMessageAge() int32
+
+	GetCachedMessageSubscriptionRequestStrategy() *CachedMessageSubscriptionStrategy
+}
+
+type cachedMessageSubscriptionRequest struct {
+	name                              string
+	cacheName                         string
+	subscription                      *TopicSubscription
+	cacheAccessTimeout                int32
+	maxCachedMessages                 int32
+	cachedMessageAge                  int32
+	cachedMessageSubscriptionStrategy *CachedMessageSubscriptionStrategy
+}
+
+// GetName retrieves the name of the topic subscription.
+func (request *cachedMessageSubscriptionRequest) GetName() string {
+	return request.name
+}
+
+// GetCacheName retrieves the name of the cache.
+func (request *cachedMessageSubscriptionRequest) GetCacheName() string {
+	return request.cacheName
+}
+
+func (request *cachedMessageSubscriptionRequest) GetSubscription() *TopicSubscription {
+	return request.subscription
+}
+func (request *cachedMessageSubscriptionRequest) GetCacheAccessTimeout() int32 {
+	return request.cacheAccessTimeout
+}
+
+func (request *cachedMessageSubscriptionRequest) GetMaxCachedMessages() int32 {
+	return request.maxCachedMessages
+}
+
+func (request *cachedMessageSubscriptionRequest) GetCachedMessageAge() int32 {
+	return request.cachedMessageAge
+}
+
+func (request *cachedMessageSubscriptionRequest) GetCachedMessageSubscriptionRequestStrategy() *CachedMessageSubscriptionStrategy {
+	return request.cachedMessageSubscriptionStrategy
+}
+
+// NewCachedMessageSubscriptionRequest returns a CachedMessageSubscriptionRequest that can be used to configure a cache request
```

Comment:
@oodigie, this doc string has not been changed and is incorrect. The updated doc string is available in the story description and arch page. Please update this.

Reply:
@TrentDaniel, just wanted to point out that we shouldn't have "ReceiverCacheRequests"; it should be singular "ReceiverCacheRequest". We don't normally name Interfaces as plurals in the Go API. I updated this in the doc string and I will add a comment on the ARCH page.

Reply:
The error in the doc string that blocks the PR was fixed. See the comment I left as a part of approving the PR for details on the interface name change. It does not block this PR.